app/scripts/controllers/network/network-controller.test.js

```diff
@@ -4639,6 +4641,110 @@ describe('NetworkController', () => {
       });
     });
   });
+  describe('upsertNetworkConfiguration', () => {
+    it('should add a network configuration', async () => {
```

Comment:
Given that this behavior is conditional, what about something like:

``` suggestion
    it('should add the given network configuration if its rpcURL does not match an existing configuration', async () => {
```

Reply:
changed [here](https://github.com/MetaMask/metamask-extension/pull/17421/commits/febce08fe121e0564df73d59772a3a5fec8fd855)